r-package/grf/R/causal_forest.R

```diff
@@ -38,6 +38,10 @@
 #' @param honesty.fraction The fraction of data that will be used for determining splits if honesty = TRUE. Corresponds
 #'                         to set J1 in the notation of the paper. When using the defaults (honesty = TRUE and
 #'                         honesty.fraction = NULL), half of the data will be used for determining splits
+#' @param prune (experimental) If true, prune the estimation sample tree such that no leaves are empty.
+#'              If false, keep the same tree as determined in the splits sample (internally, NaNs are returned for
```

Comment:
Is this true? I thought that instead of returning NaN, we make sure to skip over that tree when computing predictions.

Reply:
Sorry, I think I am wording this wrong. Yes, there is a `continue` if `leaf_node.empty()` in RegressionPredictionStrategy.cpp, I meant what percolates back to R in that case, which is a NaN:

```
library(grf)
X = matrix(rnorm(100*5), 100, 5)
Y = rnorm(100)
f = regression_forest(X, y, honesty.fraction = 0.9, prune = FALSE, num.trees = 1)
p = predict(f)$predictions
sum(is.nan(p)) / 100
[1] 0.65

```

With honesty = false, and sample.fraction=0.5, the above fraction will always be 50%

Reply:
But I think you only get `NaN` if `num.trees = 1`? I think this wording may confuse users because it could suggest that if any empty leaf is encountered, the entire estimate becomes `NaN`. We could instead say something like "(if an empty leave is encountered, that tree is skipped and does not contribute to the estimate)".